### DISCUSSION TITLE
totalUnreadMessageCount live data

### DISCUSSION THREAD

**Discussion by tarunkhatri** [GetStream/stream-chat-android#4318] [Q&A]
Hi dev team,

   implementation 'io.getstream:stream-chat-android-ui-components:5.11.0'
   implementation 'io.getstream:stream-chat-android-client:5.11.0'

i have added above libraries but i am not able to found totalUnreadMessageCount live data. 
i saw many people mention to use - 
ChatDomain.instance().totalUnreadCount.observe

above thing but this class is not available.
Please give me some solution to observe total unread message counts
thanks

**Comment by adasiewiczr** [comment]
Hey @tarunkhatri,
the `ChatDomain` was removed in the `v5` major release and `LiveData` objects were converted to `StateFlow` and moved either to `GlobalState` or `ClientState`.
You can access the total unread messages count through the `GlobalState::totalUnreadCount: StateFlow<Int>` property.
`StateFlow` can be converted to `LiveData` using [asLiveData()](https://developer.android.com/reference/kotlin/androidx/lifecycle/package-summary#(kotlinx.coroutines.flow.Flow).asLiveData(kotlin.coroutines.CoroutineContext,kotlin.Long)) extension function if you are using Kotlin or by using `FlowExtensions.asLiveData(stateFlow);` function, provided by our SDK, if you are using Java.

We are going to create more docs around using `OfflinePlugin` and I'll let you know once they are available.
Please let us know if you have any further questions.
